lib/simulator-xcode-9.3.js

```diff
@@ -22,30 +20,45 @@ class SimulatorXcode93 extends SimulatorXcode9 {
 
     // lsof -aUc launchd_sim
     // gives a set of records like:
-    //   launchd_s 69760 isaac    3u  unix 0x57aa4fceea3937f3      0t0      /private/tmp/com.apple.CoreSimulator.SimDevice.D7082A5C-34B5-475C-994E-A21534423B9E/syslogsock
-    //   launchd_s 69760 isaac    5u  unix 0x57aa4fceea395f03      0t0      /private/tmp/com.apple.launchd.2B2u8CkN8S/Listeners
-    //   launchd_s 69760 isaac    6u  unix 0x57aa4fceea39372b      0t0      ->0x57aa4fceea3937f3
-    //   launchd_s 69760 isaac    8u  unix 0x57aa4fceea39598b      0t0      /private/tmp/com.apple.launchd.2j5k1TMh6i/com.apple.webinspectord_sim.socket
-    //   launchd_s 69760 isaac    9u  unix 0x57aa4fceea394c43      0t0      /private/tmp/com.apple.launchd.4zm9JO9KEs/com.apple.testmanagerd.unix-domain.socket
-    //   launchd_s 69760 isaac   10u  unix 0x57aa4fceea395f03      0t0      /private/tmp/com.apple.launchd.2B2u8CkN8S/Listeners
-    //   launchd_s 69760 isaac   11u  unix 0x57aa4fceea39598b      0t0      /private/tmp/com.apple.launchd.2j5k1TMh6i/com.apple.webinspectord_sim.socket
-    //   launchd_s 69760 isaac   12u  unix 0x57aa4fceea394c43      0t0      /private/tmp/com.apple.launchd.4zm9JO9KEs/com.apple.testmanagerd.unix-domain.socket
-    // these _appear_ to always be grouped together (so, the records for the particular sim are all in a group, before the next sim, etc.)
-    // so starting from the correct UDID, we ought to be able to pull the next record with `com.apple.webinspectord_sim.socket` to get the correct socket
-    let {stdout} = await exec('lsof', ['-aUc', 'launchd_sim']);
-    for (let record of stdout.split('com.apple.CoreSimulator.SimDevice.')) {
-      if (!record.includes(this.udid)) {
-        continue;
-      }
-      const match = WEBINSPECTOR_SOCKET_REGEXP.exec(record);
-      if (!match) {
-        return null;
-      }
-      this.webInspectorSocket = match[1];
-      return this.webInspectorSocket;
+    //   COMMAND     PID      USER   FD   TYPE             DEVICE SIZE/OFF NODE NAME
```

Comment:
would it be possible to cover this logic with at least one simple unit test?

Reply:
thank you for your advice, added as https://github.com/appium/appium-ios-simulator/pull/352/commits/c00901a9ddea178c5581a7a57d96d8cee3f17c59

Reply:
I assume this long comments now could be reduced to a unit test link

Reply:
thanks, done as https://github.com/appium/appium-ios-simulator/pull/352/commits/adfe006c4d53d71947bc3d03b01415768f8745fc